torch/package/package_exporter.py

```diff
@@ -724,18 +722,19 @@ def _persistent_id(self, obj):
             storage_type = normalize_storage_type(type(obj))
             obj_key = str(obj._cdata)
             location = location_tag(obj)
-            name = f".data/{obj_key}.storage"
-
-            if name not in self.serialized_storages:
-                # check to see if storage was previously serialized
-                serialized_files = self.zip_file.get_all_written_records()
-                if name not in serialized_files:
-                    if obj.device.type != "cpu":
-                        obj = obj.cpu()
-                    num_bytes = obj.size() * obj.element_size()
-                    self.zip_file.write_record(name, obj.data_ptr(), num_bytes)
-                self.serialized_storages.add(name)
-            return ("storage", storage_type, obj_key, location, obj.size())
+
+            # serialize storage if not already written
+            if not self.script_module_serializer.has_storage(obj_key):
```

Comment:
We don't have to preserve the cdata naming scheme here I think. We ought to be consistent with TS and use `StorageContext` to assign names to tensors during serialization.

Reply:
I can write comments on this code, it is confusing. The cdata isn't being used for the storage name, it's just being used to identify the storage itself across cpp/python during serialization. This code is consistent with the TS code of using the `storage_id` as the filename. (Is that what your comment was asking about?)